numerical_methods/secant_method.cpp

```diff
@@ -0,0 +1,70 @@
+/**
+* \file
+* \brief Solve the equation \f$f(x)=0\f$ using
+* [Secant Method](https://en.wikipedia.org/wiki/Secant_method)
+*
+* The example function \f$f(x)\f$ looks like this:
+* \f[
+* x^4 - 29 = 0
+* \f]
+*
+* \author [Fatih Küçükkarakurt](https://github.com/fkkarakurt)
+*/
+
+/**
+* @brief Secant Method for the root of f(x)
+* Analytically the solution is
+* \f[
+* x^4-29 = 0
+* x = 29^(1/4) = 2,3205957871060836757951626724362...
+* \f]
+*/
+
+#include <cmath> /// for built-in mathematical functions
+#include <iomanip> /// to identify the manipulator and setprecision
+#include <iostream> /// for IO operations
+
+/**
+* @brief Definition the f(x) fuction
+* @param x unknown x value
+*/
+double f(double x) { return x * x * x * x - 29.0; } // The Function f(x)
+
+/**
+* @brief Main function
+* @returns 0 on exit
+*/
+int main() {
+   /**
+    * @brief Information required for the secant method
```

Comment:
Seems like the code above could be separated as a function with this documentation, IMO.